joomla/joomla-cms
# Correct issue 9509 missing index and column changes in utf8mb4 conversion

Pull Request for Issue #9509 
 .
#### Summary of Changes

Add changing of columns and indexes to utf8(mb4) conversion scripts which have been forgotten before.
- Column `user_id` of table `#__user_keys` downsized to length of 150 so it fits to column `username` of table `#__users`.
- Changed column length of column `name` in table `#__users` to 400.
- Changed follwoing indexes (keys) to include column only with 1st 100 chars:
  - TABLE `#__menu`, KEY `idx_path`
  - TABLE `#__users`, KEY `idx_name`

Column `path` of table `#__menu` has already 1024 and was not enlarged in 3.5.0's joomla.sql, and so it is also not enlarged with this PR.

Note: This PR does not solve the problem that the utf8mb4 or utf8 combersion has to be executed again after an update e.g. from 3.5.0 to the 3.5.x in which this PR will be included. This can be manually done by the SQL statement provided in testing instructions below.

A solution for that has to be developed with another PR.
#### Testing Instructions

Step 1: Apply this patch on latest staging or a 3.5.0 final release, e.g. using the patchtester.

Step 2: In phpMyAdmin, execute the SQL statement from the new schema update script included in this PR:

> UPDATE `#__utf8_conversion` SET `converted` = 0;

The reason for this step is that this PR does not solve the problem that the utf8mb4 or utf8 combersion has to be executed again after an update e.g. from 3.5.0 to the 3.5.x in which this PR will be included, see description above.

Step 3: Go to "Extensions -> Manage -> Database".

Result: Database is not up to date, utf8mb4 (or utf8) conversion is shown as to be done.

Click the "Fix" button.

Result: Database up to date.

Check with phpMyadmin if the changes mentioned in the Summary of Changes above have been applied.

Result: Changes have been applied.
